trieste/ask_tell_optimization.py

```diff
@@ -368,15 +368,30 @@ def ask(self) -> TensorType:
         #   which, when called, returns state and points
         # so code below is needed to cater for both cases
 
-        points_or_stateful = self._acquisition_rule.acquire(
-            self._search_space, self._models, datasets=self._datasets
-        )
+        with Timer() as query_point_generation_timer:
+            points_or_stateful = self._acquisition_rule.acquire(
+                self._search_space, self._models, datasets=self._datasets
+            )
 
         if callable(points_or_stateful):
             self._acquisition_state, query_points = points_or_stateful(self._acquisition_state)
         else:
             query_points = points_or_stateful
 
+        summary_writer = get_tensorboard_writer()
+        if summary_writer:
+            with summary_writer.as_default(step=get_step_number()):
+                if tf.rank(query_points) == 2:
```

Comment:
Because the type system doesn't tell me so :-)

Seriously: does it _have_ to be true (i.e. do we assume/assert it anywhere) or is just always true in practice?